### ISSUE TITLE
Can't get JDBC/Derby to work on 1.8.1

### ISSUE THREAD

**Issue by alex3kov** [openhab/openhab#4111]
I'm using examples from the Wiki. In openhab.cfg:  
```
jdbc:tableNamePrefix=Item
jdbc:tableUseRealItemNames=false
jdbc:tableIdDigitCount=0
jdbc:url=jdbc:derby:./dbDerby;create=true
jdbc:user=APP
jdbc:password=APP
```
In jdbc.persist:
```
Strategies {
    everyHour : "0 0 * * * ?"
    everyDay  : "0 0 0 * * ?"
    default = everyChange
}
Items {
    * : strategy = everyChange, everyDay, restoreOnStartup
}
```
On openhab startup database gets created in specified location (./dbDerby) and I can connect to it using SQuirrel and APP:APP user/password.  
derby-10.12.1.1.jar and org.openhab.persistence.jdbc-1.8.1.jar are present in `addons` directory.  
This is what I get in openhab.log:  
```
2016-02-28 09:14:10.953 [INFO ] [.service.AbstractActiveService] - Exec Refresh Service has been started
2016-02-28 09:14:11.349 [WARN ] [p.j.internal.JdbcConfiguration] - JDBC::updateConfig: try to load JDBC-driverClass: 'org.apache.derby.jdbc.EmbeddedDriver'
2016-02-28 09:14:11.352 [WARN ] [o.o.p.jdbc.internal.JdbcMapper] - JDBC::openConnection: setupDataSource.
2016-02-28 09:14:11.480 [WARN ] [z.hikari.util.DriverDataSource] - Registered driver with driverClassName=org.apache.derby.jdbc.EmbeddedDriver was not found, trying direct instantiation.
2016-02-28 09:14:15.503 [ERROR] [org.knowm.yank.Yank           ] - Error in SQL query!!!
java.sql.SQLException: Table/View 'ITEMS' does not exist. Query: SELECT itemid, itemname FROM items Parameters: []
        at org.apache.commons.dbutils.AbstractQueryRunner.rethrow(AbstractQueryRunner.java:392) ~[commons-dbutils-1.6.jar:1.6]
        at org.apache.commons.dbutils.QueryRunner.query(QueryRunner.java:351) ~[commons-dbutils-1.6.jar:1.6]
        at org.apache.commons.dbutils.QueryRunner.query(QueryRunner.java:289) ~[commons-dbutils-1.6.jar:1.6]
        at org.knowm.yank.Yank.queryBeanList(Yank.java:279) ~[yank-3.1.0.jar:na]
        at org.openhab.persistence.jdbc.db.JdbcBaseDAO.doGetItemIDTableNames(JdbcBaseDAO.java:238) [bundlefile:na]
        at org.openhab.persistence.jdbc.internal.JdbcMapper.getItemIDTableNames(JdbcMapper.java:85) [bundlefile:na]
        at org.openhab.persistence.jdbc.internal.JdbcMapper.checkDBSchema(JdbcMapper.java:200) [bundlefile:na]
        at org.openhab.persistence.jdbc.internal.JdbcPersistenceService.updateConfig(JdbcPersistenceService.java:218) [bundlefile:na]
        at org.openhab.persistence.jdbc.internal.JdbcPersistenceService.activate(JdbcPersistenceService.java:62) [bundlefile:na]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_95]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_95]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_95]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_95]
        at org.eclipse.equinox.internal.ds.model.ServiceComponent.activate(ServiceComponent.java:235) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.activate(ServiceComponentProp.java:146) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.model.ServiceComponentProp.build(ServiceComponentProp.java:345) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponent(InstanceProcess.java:620) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.InstanceProcess.buildComponents(InstanceProcess.java:197) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.Resolver.buildNewlySatisfied(Resolver.java:473) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.Resolver.enableComponents(Resolver.java:217) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.SCRManager.performWork(SCRManager.java:816) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.SCRManager$QueuedJob.dispatch(SCRManager.java:783) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.ds.WorkThread.run(WorkThread.java:89) [org.eclipse.equinox.ds_1.4.1.v20120926-201320.jar:na]
        at org.eclipse.equinox.internal.util.impl.tpt.threadpool.Executor.run(Executor.java:70) [org.eclipse.equinox.util_1.0.400.v20120917-192807.jar:na]
```

**Comment by watou** [comment]
@lewie, could you have a look at this please?  Thanks.

**Comment by lewie** [comment]
@watou, wondering because this never could work correctly...
@alex3kov, since this PR is not in master, you can help yourself with creating main Items Table by running this sql with SQuirrel:
`CREATE TABLE ITEMS ( ItemId INTEGER NOT NULL GENERATED ALWAYS AS IDENTITY (START WITH 1, INCREMENT BY 1), itemname VARCHAR(500) NOT NULL);`

Later you can find the working jar here: https://openhab.ci.cloudbees.com/job/openHAB1-Addons/

**Comment by alex3kov** [comment]
@lewie nope, still not working after your workaround. Attaching new error log.
[error2.txt](https://github.com/openhab/openhab/files/150545/error2.txt)

**Comment by lewie** [comment]
@alex3kov, you are right, there was a second bug.

Download the new jar for testing: https://github.com/lewie/openhab/blob/jdbc_derby_doCreateItemsTableIfNot-test/org.openhab.persistence.jdbc-1.9.0-SNAPSHOT.jar?raw=true

Thanks for testing!
Helmut

**Comment by alex3kov** [comment]
@lewie I deleted Derby database for clean testing, here's another error using your new jar:
```
2016-03-01 09:19:05.936 [INFO ] [.service.AbstractActiveService] - Exec Refresh Service has been started
2016-03-01 09:19:08.936 [INFO ] [o.o.p.jdbc.internal.JdbcMapper] - JDBC::openConnection: Driver is available::Yank setupDataSource
2016-03-01 09:19:09.071 [WARN ] [z.hikari.util.DriverDataSource] - Registered driver with driverClassName=org.apache.derby.jdbc.EmbeddedDriver was not found, trying direct instantiation.
2016-03-01 09:19:19.279 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::query: unable to find table for query, no Data in Database for Item 'Temperature'
2016-03-01 09:19:19.281 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::query: try to generate the table for item 'Temperature'
2016-03-01 09:19:20.048 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::query: unable to find table for query, no Data in Database for Item 'CO2'
2016-03-01 09:19:20.050 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::query: try to generate the table for item 'CO2'
2016-03-01 09:19:20.501 [INFO ] [runtime.busevents             ] - CO2 state updated to 782
2016-03-01 09:19:20.519 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::store: ignore Item 'All' because it is UnDefType
2016-03-01 09:19:20.531 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::store: ignore Item 'gGroundFloor' because it is UnDefType
2016-03-01 09:19:20.545 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::store: ignore Item 'gBedroom' because it is UnDefType
```
I don't see any tables being created in database (I assume it will create them in `APP`).

**Comment by lewie** [comment]
@alex3kov,
no, these are warning's (only in debug mode) and info's.
Everything works as expected now:

Driver and Library init:
````
2016-03-01 09:19:05.936 [INFO ] [.service.AbstractActiveService] - Exec Refresh Service has been started
2016-03-01 09:19:08.936 [INFO ] [o.o.p.jdbc.internal.JdbcMapper] - JDBC::openConnection: Driver is available::Yank setupDataSource
2016-03-01 09:19:09.071 [WARN ] [z.hikari.util.DriverDataSource] - Registered driver with driverClassName=org.apache.derby.jdbc.EmbeddedDriver was not found, trying direct instantiation.
````

Currently there is no table for the item 'Temperature' and 'CO2', I try to create: 
````
2016-03-01 09:19:19.279 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::query: unable to find table for query, no Data in Database for Item 'Temperature'
2016-03-01 09:19:19.281 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::query: try to generate the table for item 'Temperature'
2016-03-01 09:19:20.048 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::query: unable to find table for query, no Data in Database for Item 'CO2'
2016-03-01 09:19:20.050 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::query: try to generate the table for item 'CO2'
````

Table for item 'CO2' seams to be created, because a value is written:
````
2016-03-01 09:19:20.501 [INFO ] [runtime.busevents             ] - CO2 state updated to 782
````

The current values of the items: 'All', 'gGroundFloor' and 'gBedroom' is UnDefType (no value is set, value is empty), therefore JDBC discards the write command:
````
2016-03-01 09:19:20.519 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::store: ignore Item 'All' because it is UnDefType
2016-03-01 09:19:20.531 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::store: ignore Item 'gGroundFloor' because it is UnDefType
2016-03-01 09:19:20.545 [WARN ] [o.p.j.i.JdbcPersistenceService] - JDBC::store: ignore Item 'gBedroom' because it is UnDefType
````

**Comment by alex3kov** [comment]
@lewie IIRC, this line is logged regardless of JDBC, on every event update:
```
2016-03-01 09:19:20.501 [INFO ] [runtime.busevents             ] - CO2 state updated to 782
```
Hm, I didn't get the picture in SQuirrel that you're getting (I'm using real nodes instead of demo though) - there were no tables in APP.  
I'll recheck things and do more tests tomorrow, will post results.

**Comment by lewie** [comment]
Yes, please do that. If it does not create tables in APP please post the complete log.
Yes, CO2 was only a global busevent message, my fault.

**Comment by alex3kov** [comment]
@lewie Here's a rundown of my test:  
1. I deleted database directory (dbDerby)
2. dropped this jar https://github.com/lewie/openhab/blob/jdbc_derby_doCreateItemsTableIfNot-test/org.openhab.persistence.jdbc-1.9.0-SNAPSHOT.jar?raw=true in `addons` directory
3. started openhab, let it run for a while and stop. Starting/stopping openhab is done via this [script](https://github.com/openhab/openhab/wiki/Samples-Tricks#how-to-configure-openhab-to-start-automatically-on-linux)
4. Accessed database dbDerby via SQuirrel - no tables were created in APP.  

Attaching configs and full log. Why doesn't Github accept log/cfg files is beyond me.

![app](https://cloud.githubusercontent.com/assets/3616242/13488587/3ae2b6bc-e133-11e5-9051-076dc6253978.png)

[cfg_logs.zip](https://github.com/openhab/openhab/files/156372/cfg_logs.zip)

**Comment by lewie** [comment]
@alex3kov,

very curious!

Please can you add this lines temporarily to *[openhabfolder]/configurations/logback.xml* please.
````
	<logger name="java.sql" level="DEBUG" />
	<logger name="com.zaxxer" level="DEBUG" />
	<logger name="com.xeiam.yank" level="DEBUG" />
	<logger name="org.apache.commons" level="DEBUG" />
	<logger name="org.openhab.persistence.jdbc" level="DEBUG" />
````
So we get much more debug output. Restart, let it run and please send me openhab.log again.

Normally if a table could not be created an error is thrown. 
So we'll have to keep looking.

in addition you can test to use SQlite, we would see, if it is especially a derby problem!
Use this [jar](http://central.maven.org/maven2/org/xerial/sqlite-jdbc/3.8.11.2/sqlite-jdbc-3.8.11.2.jar)
and change url to.
````
jdbc:url=jdbc:sqlite:./sqlite.db
````
Or test with H2 or HSQLDB ([instructions](https://github.com/openhab/openhab/wiki/JDBC-Persistence)).
Incidentally, all of which are much faster than derby, see: https://github.com/openhab/openhab/pull/3248

**Comment by alex3kov** [comment]
@lewie I narrowed it down a bit: it creates tables on consecutive runs, when database directory is already in place; but fails on the first run, when database was just created. Attaching logs for both cases. I'm interested specifically in Derby - SQlite is not Java, I have no experience with H2 and I had Derby running for a couple of years, so it seems mature/stable enough. I'm willing to test with those other RDBMS after I can get Derby working, deal? :)  
[openhab_creating_DB.txt](https://github.com/openhab/openhab/files/156726/openhab_creating_DB.txt)
[openhab_existing_DB.txt](https://github.com/openhab/openhab/files/156727/openhab_existing_DB.txt)

**Comment by lewie** [comment]
@alex3kov,
we are getting closer... values are written.

The error in openhab_existing_DB.txt was now a upper-, lowercase problem.
Derby ignores case when for example writing a new table, it converts all to uppercase.
But if you read it is finicky, have to ask for *ITEMS* instead of originally generated *items*.

Download and test this new jar: 
https://github.com/lewie/openhab/blob/jdbc_derby_doCreateItemsTableIfNot-testDerby/org.openhab.persistence.jdbc-1.9.0-SNAPSHOT.jar?raw=true

**Comment by alex3kov** [comment]
@lewie That new jar didn't change table creation behaviour. After `openhab_existing_DB.txt` run on previous jar (and this jar) tables were created (despite error in the log, it seems). See below attachments with comments, maybe I wasn't clear in previous comment.  
[openhab_creating_DB_2.txt](https://github.com/openhab/openhab/files/156998/openhab_creating_DB_2.txt) - First run without dbDerby directory existing - it is created on this run due to `;create=true` connection attribute. Tables are NOT created in APP;
[openhab_existing_DB_2.txt](https://github.com/openhab/openhab/files/157001/openhab_existing_DB_2.txt) - Second run. After this, tables are in place and have some data in them.

**Comment by lewie** [comment]
@alex3kov,

Derby embedded has sometimes a known problem to opened two times/twice. So SQuirrell/openHAB can not refresh/read/write because sometimes database is locked. Be careful to connect to Derby embedded with openHAB and SQuirrell at same time.

See in in openhab_creating_DB_2.txt:
````
2016-03-03 17:52:12.518 [DEBUG] [rsistence.jdbc.db.JdbcDerbyDAO] - JDBC::doStoreItemValue sql=INSERT INTO ITEM2 (TIME, VALUE) VALUES( CURRENT_TIMESTAMP, CAST( ? as DOUBLE) ) value='700.0'
2016-03-03 17:52:12.619 [DEBUG] [o.p.j.i.JdbcPersistenceService] - JDBC: Stored item 'CO2' as '700' in SQL database at Thu Mar 03 17:52:12 MSK 2016 in 119ms.
````
No errors nothing, so the Data is written! The Tables are definitely created. If not you would get Errors.

Same in openhab_existing_DB_2.txt:
````
`2016-03-03 17:55:03.731 [DEBUG] [ersistence.jdbc.db.JdbcBaseDAO] - JDBC::storeItemValueProvider: newVal.doubleValue: '686.0'
2016-03-03 17:55:03.733 [DEBUG] [rsistence.jdbc.db.JdbcDerbyDAO] - JDBC::doStoreItemValue sql=INSERT INTO ITEM2 (TIME, VALUE) VALUES( CURRENT_TIMESTAMP, CAST( ? as DOUBLE) ) value='686.0'
2016-03-03 17:55:04.060 [DEBUG] [o.p.j.i.JdbcPersistenceService] - JDBC: Stored item 'CO2' as '686' in SQL database at Thu Mar 03 17:55:04 MSK 2016 in 346ms.
````

Test reading values from DB, make a rule like:
````
rule "TEST CO2"
when
	Time cron "0/10 * * * * ?" //10 sec
then	
	logInfo("CO2", "1 minutes ago, CO2 value was " + CO2.historicState(now.minusMinutes(1)) )
end
````

**Comment by alex3kov** [comment]
@lewie It's not possible to boot the same database in two different applications - Derby will fail with XSDB6 error. I'm connecting to openHAB's database only after shutting down openHAB completely. Just did it again - tables are not getting created on the first run (when database directory gets created) and I'm getting 
```
2016-03-03 20:42:40.239 [INFO ] [org.openhab.model.script.CO2  ] - 1 minutes ago, CO2 value was null
```
entries after openHAB was running for over 5 minutes.

**Comment by lewie** [comment]
hmm, shi....
where is OH running?
Hardware, Type, RAM...

I go from the options....

It would help me, that you test in your environment with SQlite. It is running on very much debian Linux VM's here, massive tested, stable as a rock!
For you it is same as Derby. You can touch with SQuirrell and while OH is running. ;-)

Th second thing is we should use a standard Environment. OH Demo for example. So we could exclude differences between yours and mine.

When SQlite is running well, we can go back to Derby and search again.

**Comment by alex3kov** [comment]
I'm on [Cubieboard1](http://docs.cubieboard.org/products/start#cubieboard1), running [this Debian](http://www.armbian.com/). It has been running smoothly for about a month now, 24/7.  
Attaching the logs of what I got after changing jdbc url to `jdbc:url=jdbc:sqlite:./sqlite.db`, commenting out `jdbc:user` and `jdbc:password` and dropping this [jar](http://central.maven.org/maven2/org/xerial/sqlite-jdbc/3.8.11.2/sqlite-jdbc-3.8.11.2.jar) into addons directory. It does not go well :) 
[openhab_sqlite.txt](https://github.com/openhab/openhab/files/158173/openhab_sqlite.txt)

Another thing I just noticed is that when I'm doing a second start of OpenHAB on Derby (when tables do get created) I still get these messages: 
```
[INFO ] [org.openhab.model.script.CO2  ] - 1 minutes ago, CO2 value was null
```
With OH running for over 1 minute.
> For you it is same as Derby. You can touch with SQuirrell and while OH is running. ;-)  

I don't understand that statement. Just tried to connect to OH's Derby database while OH was running - Derby didn't let me with XSDB6 error, exactly like it's supposed to.

**Comment by lewie** [comment]
Cubieboard, debian... all well. :-)
- SQlite jar well... (did you delete or move derby jar away from addons folder?)
- Please can you give me output of command *java -version* from Linux Shell?
- Openjdk should run well, but best is [Oracle Java 8](http://www.webupd8.org/2014/03/how-to-install-oracle-java-8-in-debian.html)

A new JDBC jar, had a deeper look at Derby DAO, some improvements and a fixed bug: https://github.com/lewie/openhab/blob/jdbc_derby_doCreateItemsTableIfNot-testDerby/org.openhab.persistence.jdbc-1.9.0-SNAPSHOT.jar?raw=true

For testing read from DB, please use this script:
````
rule "TEST CO2"
when
    Time cron "0/10 * * * * ?" //10 sec
then    
    logInfo("CO2", "		previousState of CO2 =" + CO2.previousState() )
end
````

Please send me the log after testing.
If this does not work I will send you a preconfigured OH.

**Comment by alex3kov** [comment]
Thanks for the thorough follow-up @lewie !  
Here's an output from `java -version`:
```
java version "1.7.0_95"
OpenJDK Runtime Environment (IcedTea 2.6.4) (7u95-2.6.4-1~deb8u1)
OpenJDK Zero VM (build 24.95-b01, mixed mode)
```
I'll do the rest of the testing tomorrow.

**Comment by alex3kov** [comment]
With that last jar, looks like we're in business! Tables are getting created on the first run (when database gets created), they are getting filled with data and logs are OK.  
Here's the last log if you're interested: [openhab.log.txt](https://github.com/openhab/openhab/files/159749/openhab.log.txt)

**Comment by lewie** [comment]
@alex3kov 

Yeahhh!

Thanks for testing, with your help we could iron out some bugs.

Then please have a look next Days. It would be helpful, when you let run a Rule like this to see if everything works as expected. Send me the log then... ;-)
@watou, after that I will make a new PR.

````
rule "Persistence Demo 10 Seconds"
when
    Time cron "0/10 * * * * ?" //10 sec
then    
		var i = Temperature_GF_Living
		//var i = Weather_Temperature
		postUpdate(i, 20.0 + (25.0 - (Math::random * 50.0).intValue) / 10.0)

		logInfo("PersistenceDemo10Seconds", " 0.		"+i.getName+" DB Values")
		
		logInfo("lastUpdate", " 1.		lastUpdate "+i.lastUpdate)
		
		var hs = i.historicState(now.minusMinutes(1), "jdbc")
		if(hs != null)
			logInfo("historicState", " 2.		historicState "+hs.state) // - Retrieves the state of the item at a certain point in time
		//else
		//	logInfo("historicState", " 2.		historicState ERROR")	
		
		var cs = i.changedSince(now.minusMinutes(1), "jdbc")
		if(cs != null)logInfo("changedSince", " 3.		changedSince "+cs)	// - Checks if the state of the item has (ever) changed since a certain point in time
		else logInfo("changedSince", " 3.		changedSince null")
		
		var us = i.updatedSince(now.minusMinutes(1), "jdbc")
		if(us != null)logInfo("updatedSince", " 4.		updatedSince "+us)	// - Checks if the state of the item has been updated since a certain point in time
		else logInfo("changedSince", " 4.		updatedSince null")
		
		var mas = i.maximumSince(now.minusMinutes(1), "jdbc")
		if(mas != null)logInfo("maximumSince", " 5.		maximumSince "+mas.state)	// - Gets the maximum value of the state of the item since a certain point in time
		else logInfo("changedSince", " 5.		maximumSince null")
		
		var mis = i.minimumSince(now.minusMinutes(1), "jdbc")
		if(mis != null)logInfo("minimumSince", " 6.		minimumSince "+mis.state)	// - Gets the minimum value of the state of the item since a certain point in time
		else logInfo("changedSince", " 6.		minimumSince null")
		
		logInfo("PersistenceDemo10Sec", " 7.		averageSince "+i.averageSince(now.minusHours(1), "jdbc")) // - Gets the average value of the state of a given item since a certain point in time.
		
		var ds = i.deltaSince(now.minusMinutes(1), "jdbc")
		logInfo("deltaSince", " 8.		deltaSince "+ds)	// - Gets the difference value of the state of a given item since a certain point in time.
		//else logInfo("changedSince", " 8.		deltaSince error")
		
		var ps = i.previousState()
		if(ps != null)logInfo("previousState", " 9.		previousState "+ps.state)	// - Retrieves the previous state of the item.
		else logInfo("changedSince", " 9.		previousState null")
		
		if(i != null)logInfo("currentState", " 10.		currentState "+i.state)	// - currentState
		else logInfo("currentState", " 9.		currentState null")
		
		//logInfo("CO2", "		previousState of CO2 =" + CO2.previousState() )

end
````

**Comment by watou** [comment]
@alex3kov @lewie great work chasing down this issue!  I look forward to a PR to formally close this issue.  Thanks very much both of you.

**Comment by alex3kov** [comment]
@lewie I've set up my OH to use your rule (fixing the item name) - it runs fine so far, I'll attach the log here in a couple of days.  
P.S. After this gets merged, I want to take a stab at getting charts working on top of JDBC.

**Comment by lewie** [comment]
Yes, perfekt @alex3kov.
Have not tested jet, but charts should (have to) work as expected with JDBC!!

**Comment by alex3kov** [comment]
@lewie But [wiki](https://github.com/openhab/openhab/wiki/Charts) says:  
> For the moment, only the server-side chart creation as PNG through rrd4j is implemented. More to come soon!

**Comment by lewie** [comment]
Ah, I understand!!!! I have a external Platform connected via MQTT/REST. Among other things, for SVG Charts I generate JSON into a OH-Text-Item by jsr223 javascript. This will be rendered by my own "multi OH Gui" coneccting via MQTT and REST. :-)


OH2 and PaperUI are a big step forward!!!
Unfortunately, Charts and the possibilities in OH GUI are partly still a lack, If I only had more time ... :-(

**Comment by lewie** [comment]
...chart creation as PNG through JDBC is implemented too!!!!

**Comment by alex3kov** [comment]
So it is :smile:  
Performance is very low though, I'm trying to find the code responsible for generating the chart to see if I can contribute an improvement.

**Comment by alex3kov** [comment]
@lewie Here's an overnight log: [openhab.log.zip](https://github.com/openhab/openhab/files/160235/openhab.log.zip)  
I started to get wrong CO2 readings with this (<30, you can see them in the log), never had this before... I'll disable JDBC and monitor the log to see if JDBC introduces this problem (unlikely, right?) or it's a  problem with something else. Will post tomorrow.

**Comment by lewie** [comment]
@watou, now PR is ready. Thanks

**Comment by teichsta** [comment]
since #4171 has been merged i assume this issue can be closed, right @lewie?

**Comment by lewie** [comment]
@teichsta, Yes!

**Comment by alex3kov** [comment]
On 21/03/16 14:47, Helmut Lehmeyer wrote:
> @teichsta <https://github.com/teichsta>, Yes!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab/issues/4111#issuecomment-199237618>
>
This fix will be in 1.8.2, right?

-- 
/--Regards, Alex/

**Comment by lewie** [comment]
@alex3kov, yes! See here, watou added it to milestone 1.8.2: https://github.com/openhab/openhab/pull/4171
